src/D3MHub.sol

```diff
@@ -0,0 +1,606 @@
+        (uint256 Art, uint256 rate,, uint256 line,) = vat.ilks(ilk);
+        require(rate == RAY, "D3MHub/rate-not-one");
+
+        ID3MPool pool = ilks[ilk].pool;
+
+        pool.preDebtChange("exec");
+        uint256 currentAssets = pool.assetBalance();
+
+        if (vat.live() == 0) {
+            // MCD caged
+            require(end.debt() == 0, "D3MHub/end-debt-already-set");
+            require(ilks[ilk].culled == 0, "D3MHub/module-has-to-be-unculled-first");
+            _unwind(
+                ilk,
+                pool,
+                type(uint256).max,
+                Mode.MCD_CAGED,
+                currentAssets
+            );
+        } else if (ilks[ilk].tic != 0 || !pool.active() || !ilks[ilk].plan.active()) {
+            // pool caged
+            _unwind(
+                ilk,
+                pool,
+                type(uint256).max,
+                ilks[ilk].culled == 1
+                ? Mode.MODULE_CULLED
+                : Mode.NORMAL,
+                currentAssets
+            );
+        } else {
+            // Determine if it needs to unwind due to debt ceilings
+            uint256 lineWad = line / RAY; // Round down to always be under the actual limit
+            uint256 Line = vat.Line();
+            uint256 debt = vat.debt();
+            uint256 toUnwind;
+            if (Art > lineWad) {
+                unchecked {
+                    toUnwind = Art - lineWad;
+                }
+            }
+            if (debt > Line) {
+                unchecked {
+                    toUnwind = _max(toUnwind, _divup(debt - Line, RAY));
+                }
+            }
+
+            // Determine if it needs to unwind due plan
+            uint256 targetAssets = ilks[ilk].plan.getTargetAssets(currentAssets);
+            if (targetAssets < currentAssets) {
+                unchecked {
+                    toUnwind = _max(toUnwind, currentAssets - targetAssets);
+                }
+            }
+
+            if (toUnwind > 0) {
+                _unwind(
+                    ilk,
+                    pool,
+                    toUnwind,
+                    Mode.NORMAL,
+                    currentAssets
+                );
+            } else {
+                uint256 toWind;
+                // All the subtractions are safe as otherwise toUnwind is > 0
+                unchecked {
+                    toWind = _min(
+                                _min(
+                                    _min(
+                                        targetAssets - currentAssets,
+                                        lineWad - Art
+                                    ),
+                                    (Line - debt) / RAY
+                                ),
+                                pool.maxDeposit() // Determine if the pool limits our total deposits
+                             );
+                }
+                require(lineWad + toWind < MAXINT256, "D3MHub/wind-overflow");
```

Comment:
It should be Art + toWind

Reply:
And probably <=

Reply:
https://github.com/makerdao/dss-direct-deposit/pull/46/commits/01ba7f71f42c18e5c01a85700fa535eca1c3550e